pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestVisitor.java

```diff
@@ -240,4 +250,14 @@ private boolean visit(PlanNode node, ServerPlanRequestContext context) {
     node.visit(this, context);
     return context.getLeafStageBoundaryNode() == null;
   }
+
+  private void applyTimestampIndex(Expression expression, PinotQuery pinotQuery) {
+    RequestUtils.applyTimestampIndex(expression, pinotQuery);
+    Function functionCall = expression.getFunctionCall();
+    if (expression.isSetFunctionCall()) {
+      for (Expression operand : functionCall.getOperands()) {
+        applyTimestampIndex(operand, pinotQuery);
+      }
+    }
+  }
```

Comment:
Yes, it is a bit strange. That is the way V1 does it. I don't completely understand why we do that on brokers instead of doing that on servers depending on whether we have the column + index or not, the same way we do that with other indexes.